contracts/RewardManagerFactory.sol

```diff
@@ -0,0 +1,216 @@
+     * @param _bonusPercentage Bonus rewards percentage for user who hasn't drawn any rewards untill endDistribution
+     * @param _burner Burner for collecting preMaturePenalty
+     * @dev deployer of contract on constructor is set as owner
+     */
+    function launchRewardManager(
+        IERC20 _cnt,
+        uint256 _startDistribution,
+        uint256 _endDistribution,
+        uint256 _upfrontUnlock,
+        uint256 _preMaturePenalty,
+        uint256 _bonusPercentage,
+        address _burner
+    ) public onlyOwner {
+        require(address(_cnt) != address(0), "Cant be Zero address");
+        require(address(_burner) != address(0), "Burner Cant be Zero address");
+
+        require(
+            _startDistribution >= block.timestamp,
+            "Start time should be greater than current"
+        ); // ideally at least 24 hours more to give investors time
+
+        require(
+            _endDistribution > _startDistribution,
+            "EndDistribution should be more than startDistribution"
+        );
+
+        RewardManager newManager = new RewardManager(
+            _cnt,
+            _startDistribution,
+            _endDistribution,
+            _upfrontUnlock,
+            _preMaturePenalty,
+            _bonusPercentage,
+            _burner
+        );
+
+        managers.push(
+            RewardManagerInfo({
+                managerAddress: address(newManager),
+                startDistribution: _startDistribution,
+                endDistribution: _endDistribution
+            })
+        ); //stacking up every crowdsale info ever made to crowdsales variable
+
+        emit RewardManagerLaunched(
+            address(newManager),
+            _startDistribution,
+            _endDistribution
+        );
+        totalRewardManagers++;
+    }
+
+    function removeRewardManager(uint256 _index) public onlyOwner {
+        require(_index <= totalRewardManagers, "Invalid Index");
+        delete managers[_index];
+    }
+
+    function userTotalVestingInfo(address _user)
+        public
+        view
+        returns (
+            uint256 totalVested,
+            uint256 totalDrawnAmount,
+            uint256 amountBurnt,
+            uint256 claimable,
+            uint256 bonusRewards,
+            uint256 stillDue
+        )
+    {
+        UserInfo memory user;
+        for (uint256 i = 0; i < totalRewardManagers; i++) {
+            address rewardManagerAddress = managers[i].managerAddress;
+            if (rewardManagerAddress != address(0)) {
+                RewardManager manager = RewardManager(rewardManagerAddress);
+                (
+                    user._totalVested,
+                    user._totalDrawnAmount,
+                    user._amountBurnt,
+                    user._claimable,
+                    user._bonusRewards,
+                    user._stillDue
+                ) = manager.vestingInfo(_user);
+                
+                if(user._totalVested) {
```

Comment:
`if ( _totalVested.sub(_totalDrawnAmount) > 0 )`

Reply:
but as this is a view function & shows user total vesting info , so despite a user has pending claimable balance or not , shouldn't that rewardManger data be used for getting his overall data

Reply:
Yes, correct, we should query and add all reward managers here.